### ISSUE TITLE
The enumeration on train_loader never stops

### ISSUE THREAD

**Issue by baowenbo** [ClementPinard/FlowNetPytorch#13]
Hi, I‘m not clear why in a single epoch, the train_loader just didn't stop to perform validation process.
I set the batch size as 8, and epoch size to be 80. Then I assume each epoch will have 10 iterations. 
But actually it just never stops (see below).
Is there any problem in    ` for i, (input, target) in enumerate(train_loader):` since we use the random sampling?
Thank you for any kind help.

`=> will save everything to mpi_sintel_both/Mon-Nov-13-12:29/flownets_bn,adam,90epochs,epochSize80,b8,lr0.0001
=> fetching img pairs in '/media/BEF0E2A7F0E2655D/MPI_Sintel/training/'
2080 samples found, 1664 train samples and 416 test samples 
=> creating model 'flownets_bn'
=> setting adam solver
Epoch: [0][0/10]	Time 31.513 (31.513)	Data 7.474 (7.474)	Loss 0.4387 (0.4387)	EPE 13.288 (13.288)
Epoch: [0][10/10]	Time 19.741 (21.515)	Data 0.000 (1.319)	Loss 0.4057 (0.4030)	EPE 13.086 (12.616)
Epoch: [0][20/10]	Time 19.748 (20.917)	Data 0.000 (0.793)	Loss 0.5682 (0.6196)	EPE 14.902 (18.513)
Epoch: [0][30/10]	Time 19.811 (20.659)	Data 0.000 (0.616)	Loss 0.3680 (0.6530)	EPE 10.569 (19.120) 
Epoch: [0][40/10]	Time 20.125 (20.616)	Data 0.000 (0.548)	Loss 0.3345 (0.5855)	EPE 10.309 (17.319) 

Epoch: [0][50/10]	Time 20.161 (20.466)	Data 0.001 (0.440)	Loss 0.3218 (0.5430)	EPE 10.909 (16.359)`

**Comment by ClementPinard** [comment]
noted, push coming soon to fix the problem.
